e2e-tests/cypress/e2e/journeys/insurance/policy-and-exports/multiple-contract-policy/multiple-contract-policy.spec.js

```diff
@@ -0,0 +1,265 @@
+      },
+    });
+
+    cy.submitInsuranceEligibilityAndStartApplication();
+
+    taskList.prepareApplication.tasks.policyTypeAndExports.link().click();
+
+    multiplePolicyField.input().click();
+    submitButton().click();
+
+    getReferenceNumber().then((id) => {
+      referenceNumber = id;
+      const expectedUrl = `${Cypress.config('baseUrl')}${INSURANCE_ROOT}/${referenceNumber}${MULTIPLE_CONTRACT_POLICY}`;
+
+      cy.url().should('eq', expectedUrl);
+    });
+  });
+
+  beforeEach(() => {
+    Cypress.Cookies.preserveOnce('_csrf');
+    Cypress.Cookies.preserveOnce('connect.sid');
+  });
+
+  it('passes the audits', () => {
+    cy.lighthouse({
+      accessibility: 100,
+      performance: 75,
+      'best-practices': 100,
+      seo: 70,
+    });
+  });
+
+  it('renders a back link with correct url', () => {
+    partials.backLink().should('exist');
+    partials.backLink().invoke('text').then((text) => {
+      expect(text.trim()).equal(LINKS.BACK);
+    });
+
+    partials.backLink().click();
+
+    const expectedUrl = `${Cypress.config('baseUrl')}${INSURANCE_ROOT}/${referenceNumber}${TYPE_OF_POLICY}`;
+
+    cy.url().should('eq', expectedUrl);
+
+    goToPageDirectly(referenceNumber);
+  });
+
+  it('renders an analytics cookies consent banner that can be accepted', () => {
+    cy.checkAnalyticsCookiesConsentAndAccept();
+  });
+
+  it('renders an analytics cookies consent banner that can be rejected', () => {
+    cy.rejectAnalyticsCookies();
+  });
+
+  it('renders a phase banner', () => {
+    cy.checkPhaseBanner();
+  });
+
+  it('renders a page title and heading with caption', () => {
+    const expectedPageTitle = `${CONTENT_STRINGS.PAGE_TITLE} - ${ORGANISATION}`;
+    cy.title().should('eq', expectedPageTitle);
+
+    headingCaption().invoke('text').then((text) => {
+      expect(text.trim()).equal(CONTENT_STRINGS.HEADING_CAPTION);
+    });
+
+    heading().invoke('text').then((text) => {
+      expect(text.trim()).equal(CONTENT_STRINGS.PAGE_TITLE);
+    });
+  });
+
+  it('renders `requested start date` label, hint and inputs', () => {
+    const fieldId = REQUESTED_START_DATE;
+    const field = multipleContractPolicyPage[fieldId];
+
+    field.label().should('exist');
+    field.label().invoke('text').then((text) => {
+      expect(text.trim()).equal(FIELDS.CONTRACT_POLICY[fieldId].LABEL);
+    });
+
+    field.hint().invoke('text').then((text) => {
+      expect(text.trim()).equal(FIELDS.CONTRACT_POLICY[fieldId].HINT);
+    });
+
+    field.dayInput().should('exist');
+    field.monthInput().should('exist');
+    field.yearInput().should('exist');
+  });
+
+  it('renders `total months of insurance` label, hint, prefix and input', () => {
+    const fieldId = TOTAL_MONTHS_OF_INSURANCE;
+    const field = multipleContractPolicyPage[fieldId];
+
+    field.label().should('exist');
+    field.label().invoke('text').then((text) => {
+      expect(text.trim()).equal(FIELDS.CONTRACT_POLICY.MULTIPLE[fieldId].LABEL);
```

Comment:
could we dereference these?

Reply:
Updated to dereference as much as possible